app/bundles/FormBundle/Model/SubmissionModel.php

```diff
@@ -795,18 +704,140 @@ function () use ($results, $translator, $name) {
                         $objWriter->save('php://output');
                     }
                 );
-                $response->headers->set('Content-Type', 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet');
-                $response->headers->set('Content-Disposition', 'attachment; filename="'.$name.'.xlsx"');
-                $response->headers->set('Expires', '0');
-                $response->headers->set('Cache-Control', 'must-revalidate');
-                $response->headers->set('Pragma', 'public');
+                $this->setResponseHeaders($response, $name.'.xlsx', [
+                    'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+                ]);
 
                 return $response;
             default:
                 return new Response();
         }
     }
 
+    /**
+     * @param array<string> $contentType
+     */
+    public function setResponseHeaders(StreamedResponse $response, string $filename, array $contentType): void
```

Comment:
Thanks for this refactoring! Could you make these methods private? They are not used and should not be used form outside. I can see it simplifies your unit tests. But methods should not be public only for testability. There are 2 approaches you can use:
1. Extract these methods to another service. In that case they must be public and you can keep the tests.
2. Make these helper methods private and test them through the public methods in this class that call them.

Reply:
Thank you! I made methods private and used ReflectionClass for tests. I think it would be nice to separate them into services in the future, but this requires good planning and more refactoring, so if you don't mind, I'd leave them in the model for now.